README.md

```diff
@@ -172,32 +183,29 @@ export interface JWTHeader {
 }
 
 export interface JWTPayload {
-  iss: string; // REQUIRED (string). The value of this claim MUST be the client_id of the client making the credential request.
+  iss?: string; // REQUIRED (string). The value of this claim MUST be the client_id of the client making the credential request.
   aud?: string; // REQUIRED (string). The value of this claim MUST be the issuer URL of credential issuer.
   iat?: number; // REQUIRED (number). The value of this claim MUST be the time at which the proof was issued using the syntax defined in [RFC7519].
   nonce?: string; // REQUIRED (string). The value type of this claim MUST be a string, where the value is a c_nonce provided by the credential issuer. //TODO: Marked as required not present in NGI flow
-  jti: string; // A new nonce chosen by the wallet. Used to prevent replay
+  jti?: string; // A new nonce chosen by the wallet. Used to prevent replay
   exp?: number; // Not longer than 5 minutes
 }
 
-export interface JWTSignerArgs {
+export interface ProofOfPossessionCallbackArgs {
+  kid: string;
   header: JWTHeader;
   payload: JWTPayload;
-  privateKey: KeyObject;
-  publicKey: KeyObject;
 }
 
 export interface ProofOfPossessionOpts {
-  issuerURL?: string;
-  clientId?: string;
-  jwtSignerArgs: JWTSignerArgs;
-  jwtSignerCallback: JWTSignerCallback;
-  jwtVerifyCallback?: JWTVerifyCallback;
+  proofOfPossessionCallback: JWTSignerCallback;
+  proofOfPossessionVerifierCallback?: JWTVerifyCallback;
+  proofOfPossessionCallbackArgs: ProofOfPossessionCallbackArgs;
```

Comment:
Why do we list the callback args as an option? I would basically expect the build/sign method to have all info present then populate the header/payload and hand it over to the registered callback to get the signed result

Reply:
Okay guess the documentation needs to be updated from actual current implementation

Reply:
changed the name of the interface from `ProofOfPossessionOpts` to `ProofOfPossessionArgs`